bayesian_cnn_prometheus/evaluation/bayesian_model_evaluator.py

```diff
@@ -0,0 +1,117 @@
+from copy import copy
+from typing import Tuple, List
+
+import nibabel as nib
+import numpy as np
+from tensorflow.python.keras.engine.training import Model
+from tqdm import tqdm
+
+from bayesian_cnn_prometheus.constants import Paths
+from bayesian_cnn_prometheus.evaluation.utils import load_nifti_file
+from bayesian_cnn_prometheus.learning.model.bayesian_vnet import bayesian_vnet
+
+
+class BayesianModelEvaluator:
+    def __init__(self, weights_path: str, input_shape: Tuple = (32, 32, 16, 1)):
+        """
+        Creates BayesianModelEvaluator object to evaluate bayesian model.
+        :param weights_path: path to bayesian model weights in h5 format
+        :param input_shape: shape of the input to the model
+        """
+        self.input_shape = input_shape
+        self.model = self.load_saved_model(weights_path)
+
+    def evaluate(self, image_path: str, samples_num: int, window: List[int]) -> List[np.array]:
+        """
+        Samples model samples_num times and returns list of predictions.
+        :param image_path: path to the image to predict
+        :param samples_num: number of samples to make
+        :param window: three-elements list with steps value to make in each axis
+        :return: list of arrays with samples_num predictions on image
+        """
+        image = load_nifti_file(image_path)
+        image_chunks, coords = self.create_chunks(image, window)
+
+        predictions = []
+        for _ in tqdm(range(samples_num)):
+            prediction = np.zeros(image.shape)
+            count_prediction = np.zeros(image.shape)
+
+            for chunk, coord in zip(image_chunks, coords):
+                reshaped_chunk = chunk.reshape(1, *chunk.shape, 1)
+                chunk_pred = self.model.predict(reshaped_chunk)
+                reshaped_chunk_pred = chunk_pred.reshape(*chunk.shape)
+
+                prediction = self.add_chunk_to_array(prediction, reshaped_chunk_pred, coord)
+                count_prediction = self.add_chunk_to_array(count_prediction, np.ones(chunk.shape), coord)
+
+            safe_count_prediction = copy(count_prediction)
+            safe_count_prediction[count_prediction == 0] = 1
+
+            predictions.append(prediction / safe_count_prediction)
+
+        return predictions
+
+    @staticmethod
+    def save_predictions(patient_id, predictions) -> None:
+        """
+        Saves predictions list in nifti file.
+        :param patient_id: four-digit patient id
+        :param predictions: list of arrays with predictions
+        """
+        img = nib.Nifti1Image(np.array(predictions), np.eye(4))
+        img.header.get_xyzt_units()
```

Comment:
To jest potrzebne?

Reply:
W sensie czy ta konkretna linijka czy zapisywanie jako nifti a nie npy?

Reply:
Konkretnie ta linijka.
Wygląda jak wywołanie funkcji która musi coś zwrócić, ale nie widzę żeby to było używane. 
Jak patrzyłem na dokumentację, to do tej funkcji była pusta